feature/interface/aggregate/otg_tests/aggregate_all_not_viable_test/aggregate_all_not_forwarding_viable_test.go

```diff
@@ -0,0 +1,1025 @@
+			t.Fatal(err)
+		}
+		// Ensure Load Balancing 50:50 on LAG_2 and LAG_3
+		weights := trafficRXWeights(t, ate, []string{agg2.ateAggName, agg3.ateAggName}, flows[0])
+		for idx, weight := range equalDistributionWeights {
+			if got, want := weights[idx], weight; got < want-ecmpTolerance || got > want+ecmpTolerance {
+				t.Errorf("ECMP Percentage for Aggregate Index: %d: got %d, want %d", idx+1, got, want)
+			}
+		}
+		if ok := verifyTrafficFlow(t, ate, flows); !ok {
+			t.Fatal("Packet Dropped, LossPct for flow ")
+		}
+	})
+
+	t.Run("RT-5.7.2.2: Setting Forwarding-Viable to False for Lag2 all ports", func(t *testing.T) {
+		// Ensure ISIS Adjacency is up on LAG_2
+		if ok := awaitAdjacency(t, dut, aggIDs[1], oc.Isis_IsisInterfaceAdjState_UP); !ok {
+			t.Fatal("ISIS Adjacency is Down on LAG_2")
+		}
+		configForwardingViable(t, dut, dutPortList[1:2], false)
+		// Ensure ISIS Adjacency is Down on LAG_2
+		if ok := awaitAdjacency(t, dut, aggIDs[1], oc.Isis_IsisInterfaceAdjState_DOWN); !ok {
+			t.Fatal("ISIS Adjacency is Established on LAG_2")
+		}
+		otgutils.WaitForARP(t, ate.OTG(), top, "IPv4")
+		startTraffic(t, dut, ate, top)
+		if err := confirmNonViableForwardingTraffic(t, dut, ate, atePortList[1:(agg2.ateLagCount+1)], dutPortList[1:(agg2.ateLagCount+1)]); err != nil {
+			t.Fatal(err)
+		}
+		// Ensure that traffic from ATE port1 to pfx4 are discarded on DUT
+		if ok := verifyTrafficFlow(t, ate, flows[1:2]); !ok {
+			t.Fatal("Packet Dropped, LossPct for flow ")
+		}
+		// Ensure there is traffic received on DUT LAG_3
+		if got := validateLag3Traffic(t, dut, ate, dutPortList[(agg2.ateLagCount+1):]); got == false {
+			t.Fatal("Packets are not Received on LAG_3")
+		}
+		if ok := verifyTrafficFlow(t, ate, flows[0:1]); !ok {
+			t.Fatal("Packet Dropped, LossPct for flow ")
+		}
+	})
+
+	t.Run("RT-5.7.2.3: Setting Forwarding-Viable to True for Lag2 one of the port", func(t *testing.T) {
+		configForwardingViable(t, dut, dutPortList[agg2.ateLagCount:(agg2.ateLagCount+1)], true)
+		startTraffic(t, dut, ate, top)
+		if err := checkBidirectionalTraffic(t, dut, dutPortList[agg2.ateLagCount:]); err != nil {
+			t.Fatal(err)
+		}
+		if err := confirmNonViableForwardingTraffic(t, dut, ate, atePortList[1:agg2.ateLagCount], dutPortList[1:agg2.ateLagCount]); err != nil {
+			t.Fatal(err)
+		}
+		if ok := verifyTrafficFlow(t, ate, flows); !ok {
+			t.Fatal("Packet Dropped, LossPct for flow ")
+		}
+	})
+}
+
+// configureDUT configures DUT
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) []string {
+
+	t.Helper()
+	fptest.ConfigureDefaultNetworkInstance(t, dut)
+	if len(dut.Ports()) < 4 {
+		t.Fatalf("Testbed requires at least 4 ports, got %d", len(dut.Ports()))
+	}
+	if len(dut.Ports()) > 4 {
+		agg2.ateLagCount = uint32(len(dut.Ports()) - 3)
+		agg3.ateLagCount = 2
+		equalDistributionWeights = []uint64{33, 67}
```

Comment:
I dont think number of members makes difference as the ECMP is verified at LAG level and both LAGs should have same traffic irrespective of number of active links .

Reply:
I have verified the same with equal/unequal number of ports in both LAG. 

I am referring below please refer section 9.2 on README. 

https://github.com/openconfig/featureprofiles/blob/main/feature/isis/otg_tests/weighted_ecmp_test/README.md

Reply:
Weighted ECMP is disabled by default (https://github.com/openconfig/public/blob/541fbd64f9a399caed2878aed344c20a93dcbad7/release/models/isis/openconfig-isis.yang#L317) and out of scope of this test; the regular ECMP shouldn't be using the link bandwidths

Reply:
Oh Ok, 

@LimeHat I haven't enable Weighted ECMP here but still it is doing Weighted ECMP by default.

Reply:
@rahulsy26 , we can go by open config model as mentioned by @LimeHat  that is by default no WECMP

Reply:
@bkreddy143 If by default no WECMP then it should have working ECMP. But it seems doing WECMP by default not sure. and Also i don't find any openconfig support to enable/disable WECMP

Reply:
@rahulsy26  as per (https://github.com/openconfig/public/blob/541fbd64f9a399caed2878aed344c20a93dcbad7/release/models/isis/openconfig-isis.yang#L317)

    leaf weighted-ecmp {
      type boolean;
      default "false";
      description
        "When set to true, all eligible multipath IS-IS routes associated with
        the instance are programmed to use weighted ECMP. An IS-IS route is
        eligible for weighted ECMP if all the next-hop interfaces in the
        multipath set have a load-balancing-weight other than 'none'.";
    }

WECMP is disable  by default and the expectation should be based on  ECMP .  Request to change the script as per ECMP and raise bug to vendor if there is no support of  WECPM enable/disable or WECMP is enabled by default.

Reply:
Done